beacon-chain/rpc/attester_server.go

```diff
@@ -27,16 +31,50 @@ func (as *AttesterServer) AttestHead(ctx context.Context, att *pbp2p.Attestation
 	return &pb.AttestResponse{AttestationHash: h[:]}, nil
 }
 
-// AttestationInfoAtSlot --
-//
-// TODO(#1505): WIP.
+// AttestationInfoAtSlot fetches the necessary information from the current canonical head
+// and beacon state for an assigned attester to perform necessary responsibilities. This includes
+// fetching the epoch boundary roots, the latest justified block root, among others.
 func (as *AttesterServer) AttestationInfoAtSlot(ctx context.Context, req *pb.AttestationInfoRequest) (*pb.AttestationInfoResponse, error) {
-	return nil, nil
-}
-
-// CrosslinkCommitteesAtSlot --
-//
-// TODO(#1505): WIP.
-func (as *AttesterServer) CrosslinkCommitteesAtSlot(ctx context.Context, req *pb.CrosslinkCommitteeRequest) (*pb.CrosslinkCommitteeResponse, error) {
-	return nil, nil
+	// Set the attestation data's beacon block root = hash_tree_root(head) where head
+	// is the validator's view of the head block of the beacon chain during the slot.
+	head, err := as.beaconDB.BlockBySlot(req.Slot)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve chain head: %v", err)
+	}
+	if head == nil {
+		return nil, fmt.Errorf("no block found at slot %d", req.Slot)
+	}
+	blockRoot, err := hashutil.HashBeaconBlock(head) // TODO(#1461): Use Tree Hashing instead (IMPORTANT).
+	if err != nil {
+		return nil, fmt.Errorf("could not tree hash beacon block: %v", err)
+	}
+	beaconState, err := as.beaconDB.State()
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch beacon state: %v", err)
+	}
+	// Fetch the epoch boundary root = hash_tree_root(epoch_boundary)
+	// where epoch_boundary is the block at the most recent epoch boundary in the
+	// chain defined by head -- i.e. the BeaconBlock where block.slot == get_epoch_start_slot(head.slot).
+	// On the server side, this is fetched by calling get_block_root(state, get_epoch_start_slot(head.slot)).
+	epochBoundaryRoot, err := blocks.BlockRoot(beaconState, helpers.StartSlot(head.Slot))
+	if err != nil {
+		return nil, fmt.Errorf("could not get epoch boundary block: %v", err)
+	}
+	// Fetch the justified block root = hash_tree_root(justified_block) where
+	// justified_block is the block at state.justified_epoch in the chain defined by head.
+	// On the server side, this is fetched by calling get_block_root(state, justified_epoch).
+	justifiedBlockRoot, err := blocks.BlockRoot(beaconState, beaconState.JustifiedEpoch)
```

Comment:
so this doesnt make sense , in the above method call for the second argument you give the boundary in terms of slots and here its in terms of epoch. It cant be both slot and epoch.

Reply:
Hey @nisdas, the previous call says:
```go
// Fetch the epoch boundary root = hash_tree_root(epoch_boundary)
// where epoch_boundary is the block at the most recent epoch boundary in the
// chain defined by head -- i.e. the BeaconBlock where block.slot == get_epoch_start_slot(head.slot).
// On the server side, this is fetched by calling get_block_root(state, get_epoch_start_slot(head.slot)).
epochBoundaryRoot, err := blocks.BlockRoot(beaconState, helpers.StartSlot(head.Slot))
if err != nil {
	return nil, fmt.Errorf("could not get epoch boundary block: %v", err)
}
```

helpers.StartSlot == get_epoch_start_slot, so it is indeed in terms of epochs, just the naming can be a bit weird